include/ginkgo/core/base/index_set.hpp

```diff
@@ -345,11 +462,11 @@ class index_set {
                           const bool is_sorted);
 
     std::shared_ptr<const Executor> exec_;
-    index_type index_space_size_;
-    index_type num_stored_indices_;
-    gko::Array<index_type> subsets_begin_;
-    gko::Array<index_type> subsets_end_;
-    gko::Array<index_type> superset_cumulative_indices_;
+    index_type index_space_size_{};
+    index_type num_stored_indices_{};
+    gko::Array<index_type> subsets_begin_{};
+    gko::Array<index_type> subsets_end_{};
+    gko::Array<index_type> superset_cumulative_indices_{};
```

Comment:
I don't think we need this, since all remaining constructors initialize them already!